incubator/hnc/pkg/kubectl/tree.go

```diff
@@ -33,18 +35,33 @@ var (
 
 var treeCmd = &cobra.Command{
 	Use:   "tree",
-	Short: "Displays the hierarchy tree rooted at the given namespace",
-	Args:  cobra.ExactArgs(1),
+	Short: "Display one or more hierarchy trees",
 	Run: func(cmd *cobra.Command, args []string) {
-		nnm := args[0]
+		flags := cmd.Flags()
 		footnotesByMsg = map[string]int{}
 		footnotes = []string{}
-		hier := getHierarchy(nnm)
-		fmt.Println(nameAndFootnotes(hier))
-		printSubtree("", hier)
-
+		var nsList []string
+		defaultList := len(args) == 0
+		nsList = args
+		if flags.Changed("all-namespaces") {
+			nsList = getAllNamespaces()
+		}
+		if defaultList && !flags.Changed("all-namespaces") {
+			fmt.Printf("Error: Must specify the root of the tree(s) to display or else specify --all-namespaces\n")
+			return
```

Comment:
This should probably be `os.Exit(1)` to indicate to any calling script that there was an error.